packages/editor/src/components/block-settings-menu/content-only-settings-menu.js

```diff
@@ -19,53 +20,84 @@ import { store as editorStore } from '../../store';
 import { unlock } from '../../lock-unlock';
 
 function ContentOnlySettingsMenuItems( { clientId, onClose } ) {
-	const { entity, onNavigateToEntityRecord } = useSelect(
-		( select ) => {
-			const {
-				getBlockEditingMode,
-				getBlockParentsByBlockName,
-				getSettings,
-				getBlockAttributes,
-			} = select( blockEditorStore );
-			const contentOnly =
-				getBlockEditingMode( clientId ) === 'contentOnly';
-			if ( ! contentOnly ) {
-				return {};
-			}
-			const patternParent = getBlockParentsByBlockName(
-				clientId,
-				'core/block',
-				true
-			)[ 0 ];
-
-			let record;
-			if ( patternParent ) {
-				record = select( coreStore ).getEntityRecord(
-					'postType',
-					'wp_block',
-					getBlockAttributes( patternParent ).ref
+	const { entity, onNavigateToEntityRecord, canEditTemplateParts } =
+		useSelect(
+			( select ) => {
+				const {
+					getBlockEditingMode,
+					getBlockParentsByBlockName,
+					getSettings,
+					getBlockAttributes,
+				} = select( blockEditorStore );
+				const contentOnly =
+					getBlockEditingMode( clientId ) === 'contentOnly';
+				const _canEditTemplateParts = select( coreStore ).canUser(
+					'create',
+					'templates'
 				);
-			} else {
-				const { getCurrentPostType, getCurrentTemplateId } =
-					select( editorStore );
-				const currentPostType = getCurrentPostType();
-				const templateId = getCurrentTemplateId();
-				if ( currentPostType === 'page' && templateId ) {
+				if ( ! contentOnly ) {
+					return {
+						canEditTemplateParts: _canEditTemplateParts,
+					};
+				}
+				const patternParent = getBlockParentsByBlockName(
+					clientId,
+					'core/block',
+					true
+				)[ 0 ];
+
+				let record;
+				if ( patternParent ) {
 					record = select( coreStore ).getEntityRecord(
 						'postType',
-						'wp_template',
-						templateId
+						'wp_block',
+						getBlockAttributes( patternParent ).ref
 					);
+				} else {
+					const { getCurrentTemplateId } = select( editorStore );
+					const templateId = getCurrentTemplateId();
+					if ( templateId ) {
+						record = select( coreStore ).getEntityRecord(
+							'postType',
+							'wp_template',
+							templateId
+						);
+					}
 				}
-			}
-			return {
-				entity: record,
-				onNavigateToEntityRecord:
-					getSettings().onNavigateToEntityRecord,
-			};
-		},
-		[ clientId ]
-	);
+				return {
+					canEditTemplateParts: _canEditTemplateParts,
+					entity: record,
+					onNavigateToEntityRecord:
+						getSettings().onNavigateToEntityRecord,
+				};
+			},
+			[ clientId ]
+		);
+
+	// The post has a parent template part, but the user does not have the capability to edit template parts.
+	if ( ! canEditTemplateParts && entity ) {
```

Comment:
This needs better testing, I'm worried about unknown-to-me scenarios.

It should only show where the post has a template and the user can't edit said template.